system/jlib/jlog.cpp

```diff
@@ -2391,22 +2391,25 @@ void setupContainerizedLogMsgHandler()
         if (logConfig->hasProp(logFormatAtt))
         {
             const char *logFormat = logConfig->queryProp(logFormatAtt);
-            bool newFormatDetected = false;
+            HandleLogMsgHandler *previousStderrHandler{nullptr};
             if (streq(logFormat, "xml"))
             {
+                previousStderrHandler = theStderrHandler;
                 theStderrHandler = new HandleLogMsgHandlerXML(stderr, MSGFIELD_STANDARD);
-                newFormatDetected = true;
             }
             else if (streq(logFormat, "json"))
             {
+                previousStderrHandler = theStderrHandler;
                 theStderrHandler = new HandleLogMsgHandlerJSON(stderr, MSGFIELD_STANDARD);
-                newFormatDetected = true;
             }
             else if (!streq(logFormat, "table"))
                 LOG(MCoperatorWarning, "JLog: Invalid log format configuration detected '%s'!", logFormat);
 
-            if (newFormatDetected)
+            if (previousStderrHandler)
+            {
                 theManager->resetMonitors();
+                delete previousStderrHandler;
```

Comment:
not new really (existing deletes in MODULE_EXIT), but the handler is an IInterface (link counted object) and should be released, not deleted,
but given previous comment, this point is moot.